### ISSUE TITLE
Set up continues deployment to Digitalocean kubernetes cluster

### ISSUE THREAD

**Issue by maidul98** [Infisical/infisical#83]
It would be really nice to have some github workflow or some other workflow to streamline our deployment. Ideally, we would have a `dev` stage  and a `prod` stage. The workflow should enable us roll back quickly and etc. This is quite a pain at the moment as we have to deploy manually. Having this set up would enable us to ship code fatser and more reliably.

**Comment by samyakpiya** [comment]
Hi @maidul98 ! I'm interested in working on this issue. To ensure I understand the requirements correctly and can provide the most value, could you please clarify:

1. Current Setup:
   - Is there a specific part of the application that needs to be deployed to DigitalOcean K8s? (frontend, backend, or both?)
   - Do you already have a DigitalOcean K8s cluster set up, or should the workflow include cluster provisioning?
   - Are there any existing Kubernetes manifests for the application?

2. Deployment Requirements:
   - For the dev/prod stages, should this mirror the current AWS pipeline's approach (gamma → prod-us → prod-eu), or do you envision a different flow?
   - Should the workflow be triggered automatically on merges to main (like typical CD) or manually (like the current `workflow_dispatch`)?
   - For rollbacks, what's the preferred strategy? (Previous image versions, helm rollback, etc.)

3. Integration Points:
   - Should this integrate with your existing Kubernetes Operator?
   - Do you want to use Helm for package management?
   - Are there any specific secrets or configurations that need to be managed differently in the DO K8s environment?

I see you already have a robust AWS ECS deployment pipeline. Would you like this new workflow to:
1. Replace the existing AWS pipeline
2. Run in parallel with AWS (multi-cloud deployment)
3. Start as a separate workflow and gradually migrate from AWS to DO

Happy to help once I have a better understanding of these points! 🚀

**Comment by maidul98** [comment]
Hey @samyakpiya, Infisical's infrastructure has changed vastly and this issue is no longer relevant.
